packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,326 @@
+    }
+
+    const sqlFirst = sql.raw(pagination?.first ? pagination.first.toString() : '25');
+    const sqlAfter = sql.raw(pagination?.after ? pagination.after.toString() : '0');
+
+    const where: SqlValue[] = [];
+    where.push(sql`organization_id = ${organizationSlug}`);
+
+    if (filter?.startDate && filter?.endDate) {
+      const from = this.formatToClickhouseDateTime(filter.startDate.toISOString());
+      const to = this.formatToClickhouseDateTime(filter.endDate.toISOString());
+      where.push(sql`timestamp >= ${from} AND timestamp <= ${to}`);
+    }
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT *
+        FROM audit_logs
+        ${whereClause}
+        ORDER BY timestamp DESC
+        LIMIT ${sqlFirst}
+        OFFSET ${sqlAfter}
+      `,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async uploadToS3(
+    data: string,
+    key: string,
+  ): Promise<{ ok: { url: string } } | { error: { message: string } }> {
+    const s3Storage = this.s3Config[0];
+    const { endpoint, bucket, client } = s3Storage;
+
+    const uploadResult = await client.fetch([endpoint, bucket, key].join('/'), {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'text/csv',
+      },
+      body: data,
+    });
+
+    if (!uploadResult.ok) {
+      return {
+        error: {
+          message: 'Failed to upload the file',
+        },
+      };
+    }
+
+    const getPresignedUrl = await client.fetch([endpoint, bucket, key].join('/'), {
+      method: 'GET',
+      aws: {
+        signQuery: true,
+      },
+    });
+
+    if (!getPresignedUrl.ok) {
+      return {
+        error: {
+          message: 'Failed to get the pre-signed URL',
+        },
+      };
+    }
+
+    return {
+      ok: {
+        url: getPresignedUrl.url,
+      },
+    };
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: Date | null; endDate?: Date | null },
+  ): Promise<{ ok: { url: string } | null } | { error?: { message: string } | null }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      return {
+        error: {
+          message: 'Unauthorized: You are not authorized to perform this action',
+        },
+        ok: null,
+      };
+    }
+
+    const formattedStartDate = filter?.startDate ? new Date(filter.startDate) : undefined;
+    const formattedEndDate = filter?.endDate ? new Date(filter.endDate) : undefined;
+
+    try {
+      const { email } = await this.session.getViewer();
+      const getAllAuditLogs = await this.getPaginatedAuditLogs(
+        organizationSlug,
+        { first: 1000, after: 0 },
```

Comment:
TODO @TuvalSimha Why limit the amount to 1000 items? We want to include all the items of the time range.